### PR TITLE
update the configuration to use python 3.8 for edx-platform

### PR DESCRIPTION
**Pull request by mraarif**
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR if it is code shared across multiple services and you don't own all of the services.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).
  - [ ] Think about how this change will affect Open edX operators.  Have you updated the wiki page for the next Open edX release?

### CODE REVIEW

File: playbooks/roles/edxapp/tasks/main.yml [hunk 1]
```diff
@@ -152,6 +152,19 @@
     - install
     - install:base
 
+- name: add deadsnakes repo
+  apt_repository:
+      repo: ppa:deadsnakes/ppa
```
Comment (morenol) [inline playbooks/roles/edxapp/tasks/main.yml:157]: I think that we should add this repo if we are not in ubuntu focal, we could add that condition to this task.

Comment (morenol) [reply playbooks/roles/edxapp/tasks/main.yml:157]: I am sorry, I just noticed that this is a draft

**Comment by mraarif** [comment]
I've filed an SRE ticket here: https://openedx.atlassian.net/browse/DOS-1555 to review these changes

**Comment by mraarif** [comment]
there's a sandbox here: https://tools-edx-jenkins.edx.org/job/sandboxes/job/CreateSandbox/33291/ that built successfully using this branch

File: playbooks/roles/edxapp/tasks/main.yml (hunk 1 above)
Comment (jdmulloy) [reply playbooks/roles/edxapp/tasks/main.yml:157]: I think just adding deadsnakes all the time is easier. If we use the Ubnuntu repo the package name is ```python3```. Focal will likely be stuck on 3.8 forever and we'll be adding deadsnakes to get newer python versions, so I think it's simpler to just always install it and install the packages for the specific versions of python we want.

**Review by jdmulloy — APPROVED** [review]

**Review by adzuci — APPROVED** [review]
LGTM

**Pull request merged**